eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -87,46 +87,40 @@ case class Commitments(channelId: ByteVector32,
 
   import Commitments._
 
-  require(channelFeatures.paysDirectlyToWallet == localParams.walletStaticPaymentBasepoint.isDefined, s"localParams.walletStaticPaymentBasepoint must be defined only for commitments that pay directly to our wallet (channel features: $channelFeatures")
+  require(channelFeatures.paysDirectlyToWallet == localParams.walletStaticPaymentBasepoint.isDefined, s"localParams.walletStaticPaymentBasepoint must be defined only for commitments that pay directly to our wallet (channel features: $channelFeatures)")
   require(channelFeatures.hasFeature(DualFunding) == localParams.requestedChannelReserve_opt.isEmpty, "custom local channel reserve is incompatible with dual-funded channels")
   require(channelFeatures.hasFeature(DualFunding) == remoteParams.requestedChannelReserve_opt.isEmpty, "custom remote channel reserve is incompatible with dual-funded channels")
+  if (channelFeatures.hasFeature(Features.UpfrontShutdownScript)) require(localParams.upfrontShutdownScript_opt.isDefined, s"localParams.upfrontShutdownScript_opt must be defined for commitments that set upfront_shutdown_script (channel features: $channelFeatures)")
```

Comment:
Consistency with other `require`s. Also should be moved between static payment basepoint and dual funding requirements.

```suggestion
  require(channelFeatures.hasFeature(UpfrontShutdownScript) == localParams.upfrontShutdownScript_opt.isDefined, s"localParams.upfrontShutdownScript_opt must be defined for commitments that set upfront_shutdown_script (channel features: $channelFeatures)")
```

Reply:
done in https://github.com/ACINQ/eclair/pull/2565/commits/a3da92a3f0bac4f0d093ce5548eb89f5b1287ed1 but with a different check as old DATA_CLOSING may include data where localParams.upfrontShutdownScript_opt is defined but `upfront_shutdownscript` is not set)

Reply:
Right, what we should really do is revisit and simplify this require when we do the follow-up PR that cleans up redundant scripts.